src/core_editor/line_buffer.rs

```diff
@@ -298,6 +240,10 @@ impl LineBuffer {
         }
     }
 
+    pub fn word_count(&self) -> usize {
+        self.lines.concat().trim().split(' ').count()
+    }
```

Comment:
For readability sake, I extracted this out. Though the exact name that is given here is a bit misleading as it does not handle all such cases and I have added a failing test that showcases this. Let me know what do we want to do here. 
Here are a few ways that I see we can tackle this scenario.

- Is this a worthwhile abstraction then I focus on fixing this using the test as a guide
- it isn't a worthwhile abstraction then we can just simply use it's internal detail at the caller's site. Given this is just an expression just replacing it as is should be sufficient.
- We don't worry about it as much at let is stay as is. (and open a good-first-issue to fix this)

Reply:
You could use `split_whitespace` instead of `.split(' ')` and I think that should be enough to pass the test you mention.